tests/app/forms/test_date_form.py

```diff
@@ -52,6 +53,85 @@ def test_generate_year_date_form_creates_empty_form(self):
         self.assertFalse(hasattr(form, 'month'))
         self.assertTrue(hasattr(form, 'year'))
 
+    def test_date_form_empty_data(self):
+        schema = load_schema_from_params('test', 'dates')
+        error_messages = schema.error_messages
+
+        answers = schema.get_answers_by_id_for_block('date-block')
+
+        with self.test_request_context('/'):
+            form = get_date_form(AnswerStore(), {}, answers['single-date-answer'], error_messages=error_messages)
+
+        self.assertIsNone(form().data)
+
+    def test_month_year_date_form_empty_data(self):
+        schema = load_schema_from_params('test', 'dates')
+        error_messages = schema.error_messages
+
+        answers = schema.get_answers_by_id_for_block('date-block')
+
+        with self.test_request_context('/'):
+            form = get_month_year_form(answers['month-year-answer'], {}, {}, error_messages=error_messages)
+
+        self.assertIsNone(form().data)
+
+    def test_year_date_form_empty_data(self):
+        schema = load_schema_from_params('test', 'dates')
+        error_messages = schema.error_messages
+
+        answers = schema.get_answers_by_id_for_block('date-block')
+
+        form = get_year_form(answers['year-date-answer'], {}, {}, error_messages=error_messages, label=None, guidance=None)
+
+        self.assertIsNone(form().data)
+
+    def test_date_form_format_data(self):
+        schema = load_schema_from_params('test', 'dates')
+        error_messages = schema.error_messages
+
+        answers = schema.get_answers_by_id_for_block('date-block')
+
+        data = {'field': '2000-01-01'}
```

Comment:
Just a little thing (my little OCD regarding test dates) but can the DD field be greater than 12. Shows and tests clearly that the date format is being correctly used throughout.